src/main/java/com/rackspace/ceres/app/services/DataWriteService.java

```diff
@@ -138,4 +150,38 @@ private void cleanTags(Map<String, String> tags) {
         .retryWhen(appProperties.getRetryInsertDownsampled().build())
         .checkpoint();
   }
+
+  private Mono<?> storeMetricGroup(Metric metric) {
+    String metricGroup = metric.getMetric();
+    String metricName = metric.getMetric();
+    if(metric.getMetric().contains("_"))  {
+      metricGroup = metric.getMetric().substring(0, metric.getMetric().indexOf("_"));
```

Comment:
Can we simply this by using String.split method

Reply:
as per logic, string just before UNDERSCORE will be used as metric_group while rest of the string will be still used as metricName. So, thought of separating metric with UNDERSCORE using substring.